### PR TITLE
Fix error with no application in edge

### PR DESCRIPTION
**Pull request by tomaszzmuda**
There is a problem when we try to open new window with "about:blank" in url. We get error: 
"You'll need a new app to open this about" with link to the app store with empty list of applications supports this.

image

I used it to wrote an Add-In to Ms Office products so popup window in edge/internet explorer is the only option to login.

What have I changed:
Move a bit lines with opening popup and navigate directly to login page, not about:blank.

### CODE REVIEW

**Comment by jasonnutter** [comment]
Yeah I'll take a look and test this out.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 1]
```diff
@@ -538,14 +528,22 @@ export class UserAgentApplication {
...
+      }
+      else {
+        // prompt user for interaction
+        this.navigateWindow(urlNavigate, popUpWindow);
+      }
     }).catch((err) => {
       this.logger.warning("could not resolve endpoints");
       this.authErrorHandler(interactionType, ClientAuthError.createEndpointResolutionError(err.toString), buildResponseStateOnly(request.state), reject);
-      if (popUpWindow) {
-        popUpWindow.close();
```
Comment (jasonnutter) [inline lib/msal-core/src/UserAgentApplication.ts]: Why was this removed?

**Comment by jasonnutter** [comment]
@tomaszzmuda I'm not able to reproduce this on my machine, as Edge is able to open the pop up with `about:blank` with no issue. As the issue templates asks, can you please provide more details about how to reproduce this (e.g. you mention you are making an office add-in)? Thanks!

File: lib/msal-core/src/UserAgentApplication.ts [hunk 2]
```diff
@@ -538,14 +528,22 @@ export class UserAgentApplication {
...
 
-      // prompt user for interaction
-      this.navigateWindow(urlNavigate, popUpWindow);
+      let popUpWindow: Window;
+      if (interactionType === Constants.interactionTypePopup) {
+        // Generate a popup window
+        popUpWindow = this.openWindow(urlNavigate, "_blank", 1, this, resolve, reject);
+        if (!popUpWindow) {
+          // We pass reject in openWindow, we reject there during an error
+          return;
```
Comment (jasonnutter) [inline lib/msal-core/src/UserAgentApplication.ts]: My understanding is that this code is attempting to open a popup to a blank page, to check if popups are allowed (i.e. `popUpWindow` will be `null` if popups are blocked), and then short-circuits the `acquireTokenHelper` function if that happens. This ensures none that `acquireTokenAuthority.resolveEndpointsAsync` and the code in its `.then` don't run unnecessarily. I'm assuming this is done intentionally, to prevent orphaned callbacks (which are registered above).

Comment (tomaszzmuda) [reply lib/msal-core/src/UserAgentApplication.ts]: If the browser cannot open the link, the code hangs on line with "openWindow".

File: lib/msal-core/src/UserAgentApplication.ts (hunk 1 above)
Comment (tomaszzmuda) [reply lib/msal-core/src/UserAgentApplication.ts]: I assume that was added to close popUp if some of the previous actions fail (access to cache, build url). Right now, show popup is the last action so if popUp won't open it doesn't have to be closed.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 2 above)
Comment (jasonnutter) [reply lib/msal-core/src/UserAgentApplication.ts]: > If the browser cannot open the link, the code hangs on line with "openWindow".

Specifically in the scenario you describe above, correct? I'm referring to scenarios such as the browser having popups blocked, for example.

Comment (tomaszzmuda) [reply lib/msal-core/src/UserAgentApplication.ts]: Yes. I didn't find any other scenarios when it doesn't work. I still didn't find why it hangs ;/

**Comment by pkanher617** [comment]
@tomaszzmuda I haven't seen this issue on Edge on my devices either. Is this happening with a specific version of Edge? I would also recommend filing an issue, as we can track it as a bug rather than a PR.

**Comment by jasonnutter** [comment]
@pkanher617 I have seen other issues related to `about:blank` (such as #1006), so it might be worth seeing if we can determine why we navigate to `about:blank` first, and if it still needed, given it seems to cause issues in certain situations.

**Comment by jasonnutter** [comment]
Closing, as this has been addressed.
